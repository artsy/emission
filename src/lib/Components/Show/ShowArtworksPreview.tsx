import { Box, Serif } from "@artsy/palette"
import { ShowArtworksPreview_show } from "__generated__/ShowArtworksPreview_show.graphql"
import GenericGrid from "lib/Components/ArtworkGrids/GenericGrid"
import { CaretButton } from "lib/Components/Buttons/CaretButton"
import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

interface Props {
  onViewAllArtworksPressed: () => void
  show: ShowArtworksPreview_show
  title: string
}

export class ShowArtworksPreview extends React.Component<Props> {
  render() {
    const { show, onViewAllArtworksPressed, title } = this.props
    if (!show) {
      return null
    }
    const { artworks, counts } = show
    return (
      <>
        {title && (
          <Serif size="6" mt={2} mb={3}>
            {title}
          </Serif>
        )}
        <GenericGrid artworks={this.props.show.artworks} />
        {counts &&
          counts.artworks > artworks.length && (
            <Box mt={1}>
              <CaretButton text={`View all ${artworks.length} works`} onPress={() => onViewAllArtworksPressed()} />
            </Box>
          )}
      </>
    )
  }
}

export const ShowArtworksPreviewContainer = createFragmentContainer(ShowArtworksPreview, {
  show: graphql`
    fragment ShowArtworksPreview_show on Show {
      __id
      artworks(size: 6) {
        ...GenericGrid_artworks
      }
      counts {
        artworks
      }
    }
  `,
})
