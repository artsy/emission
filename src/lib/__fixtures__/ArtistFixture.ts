export const ArtistFixture = {
  id: "pablo-picasso",
  __id: "QXJ0aXN0OnBhYmxvLXBpY2Fzc28=",
  name: "Pablo Picasso",
  is_followed: false,
  nationality: "Spanish",
  birthday: "1881",
  deathday: "1973",
  image: {
    url: "https://d32dm0rphc51dk.cloudfront.net/i3rCA3IaKE-cLBnc-U5swQ/tall.jpg",
  },
}
