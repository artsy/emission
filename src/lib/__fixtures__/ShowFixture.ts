export const ShowFixture = {
  id: "anderson-fine-art-gallery-flickinger-collection",
  location: {
    __id: "TG9jYXRpb246NWIzNjQ2NjRjYjRjMjcxNzQzNDkwMTkx",
    id: "5b364664cb4c271743490191",
    city: "Sea Island ",
    address: "Online Exclusive",
    address_2: "",
    coordinates: null,
    day_schedules: [],
  },
  artists: [
    {
      __id: "QXJ0aXN0OmhhbnMtaG9mbWFubg==",
      id: "hans-hofmann",
      name: "Hans Hofmann",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmRhdmlkLWhvY2tuZXk=",
      id: "david-hockney",
      name: "David Hockney",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnNhbS1mcmFuY2lz",
      id: "sam-francis",
      name: "Sam Francis",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnBhYmxvLXBpY2Fzc28=",
      id: "pablo-picasso",
      name: "Pablo Picasso",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OndvbGYta2Fobg==",
      id: "wolf-kahn",
      name: "Wolf Kahn",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnJvYmVydC1tb3RoZXJ3ZWxs",
      id: "robert-motherwell",
      name: "Robert Motherwell",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmppbS1kaW5l",
      id: "jim-dine",
      name: "Jim Dine",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmVsbHN3b3J0aC1rZWxseQ==",
      id: "ellsworth-kelly",
      name: "Ellsworth Kelly",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmJyaWRnZXQtcmlsZXk=",
      id: "bridget-riley",
      name: "Bridget Riley",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmNhcnJvbGwtZHVuaGFt",
      id: "carroll-dunham",
      name: "Carroll Dunham",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnJpY2hhcmQtZGllYmVua29ybg==",
      id: "richard-diebenkorn",
      name: "Richard Diebenkorn",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmthcmVsLWFwcGVs",
      id: "karel-appel",
      name: "Karel Appel",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmRhbGUtY2hpaHVseQ==",
      id: "dale-chihuly",
      name: "Dale Chihuly",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnBvbGx5LWFwZmVsYmF1bQ==",
      id: "polly-apfelbaum",
      name: "Polly Apfelbaum",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmNhcm9seW4tYnJhZHk=",
      id: "carolyn-brady",
      name: "Carolyn Brady",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0Om5lbGwtYmxhaW5l",
      id: "nell-blaine",
      name: "Nell Blaine",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OmJydWNlLWJlYXNsZXk=",
      id: "bruce-beasley",
      name: "Bruce Beasley",
      is_followed: false,
    },
    {
      __id: "QXJ0aXN0OnJvYmVydC1oYXJtcw==",
      id: "robert-harms",
      name: "Robert Harms",
      is_followed: false,
    },
  ],
  artworks: [
    {
      __id: "QXJ0d29yazpkYXZpZC1ob2NrbmV5LXBhcGVyLXBvb2xz",
      id: "david-hockney-paper-pools",
      artist_names: "David Hockney",
      image: {
        id: "5b3a33267622dd0a9cfcfdc4",
        url: "https://d32dm0rphc51dk.cloudfront.net/banGNBZu7AfvFoXMz2zXew/normalized.jpg",
      },
      price: "",
      availability: "sold",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXN0b3VyaGVhZC1hY3Jvc3MtdGhlLWxha2U=",
      id: "carolyn-brady-stourhead-across-the-lake",
      artist_names: "Carolyn Brady",
      image: {
        id: "5b4b636f64067502ce65a990",
        url: "https://d32dm0rphc51dk.cloudfront.net/tH19ZfyCDocnNsKzXOZfHg/normalized.jpg",
      },
      price: "$15,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpyb2JlcnQtaGFybXMtYmx1ZmYtcm9hZA==",
      id: "robert-harms-bluff-road",
      artist_names: "Robert Harms",
      image: {
        id: "5b33ebad275b240e5f927a93",
        url: "https://d32dm0rphc51dk.cloudfront.net/xU1Y9RXzBJ0TY6G0ZvbHHw/normalized.jpg",
      },
      price: "$18,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpqaW0tZGluZS1hbi1pbmZvcm1hbC10aWU=",
      id: "jim-dine-an-informal-tie",
      artist_names: "Jim Dine",
      image: {
        id: "5b33e310cb4c273c8abed039",
        url: "https://d32dm0rphc51dk.cloudfront.net/dBBr15AyJ13x5heIyctJIA/tall.jpg",
      },
      price: "$20,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpyaWNoYXJkLWRpZWJlbmtvcm4tYmx1ZS13aXRoLXJlZC0xNQ==",
      id: "richard-diebenkorn-blue-with-red-15",
      artist_names: "Richard Diebenkorn",
      image: {
        id: "5b33d87dcd530e5746e1187b",
        url: "https://d32dm0rphc51dk.cloudfront.net/txGh8F3S_CDBKt6QPtjIBw/normalized.jpg",
      },
      price: "$35,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpjYXJyb2xsLWR1bmhhbS1jb2xvci1tZXNzYWdlcy1hLWItYW5kLWM=",
      id: "carroll-dunham-color-messages-a-b-and-c",
      artist_names: "Carroll Dunham",
      image: {
        id: "5b33ea119c18db46398bf3a6",
        url: "https://d32dm0rphc51dk.cloudfront.net/mfUASgYUqCEzUXsZlu6m5g/normalized.jpg",
      },
      price: "$4,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazprYXJlbC1hcHBlbC1oYXBweS1naG9zdA==",
      id: "karel-appel-happy-ghost",
      artist_names: "Karel Appel",
      image: {
        id: "5b33c803a09a6779c76cd6af",
        url: "https://d32dm0rphc51dk.cloudfront.net/SUPKae4rRR1ZWs5mc70QJg/normalized.jpg",
      },
      price: "$6,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpuZWxsLWJsYWluZS1hdWd1c3QtbGlnaHQ=",
      id: "nell-blaine-august-light",
      artist_names: "Nell Blaine",
      image: {
        id: "5b33cf9b7076d070a64d7ac7",
        url: "https://d32dm0rphc51dk.cloudfront.net/P_4Avd1FcPjMKJ_aJFA1WQ/normalized.jpg",
      },
      price: "$10,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpicnVjZS1iZWFzbGV5LXZveWFnZXI=",
      id: "bruce-beasley-voyager",
      artist_names: "Bruce Beasley",
      image: {
        id: "5babb3f7c786fb5e3fc9f10f",
        url: "https://d32dm0rphc51dk.cloudfront.net/m7N4jeFJJr3FdFqU_0YDRg/normalized.jpg",
      },
      price: "$30,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpzYW0tZnJhbmNpcy11bmRlci1ibHVlLTM=",
      id: "sam-francis-under-blue-3",
      artist_names: "Sam Francis",
      image: {
        id: "5b33ea9e5b2fba17ddc57257",
        url: "https://d32dm0rphc51dk.cloudfront.net/gz2wa9iBcZKB0ut6gMED1w/normalized.jpg",
      },
      price: "$12,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpjYXJvbHluLWJyYWR5LXRhbWFyaW5k",
      id: "carolyn-brady-tamarind",
      artist_names: "Carolyn Brady",
      image: {
        id: "5b33d4caa6ca6d137cebf02b",
        url: "https://d32dm0rphc51dk.cloudfront.net/tkm-BtRS1igRoRUx5SdOOA/normalized.jpg",
      },
      price: "$4,500",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpyb2JlcnQtbW90aGVyd2VsbC1udW1iZXItOA==",
      id: "robert-motherwell-number-8",
      artist_names: "Robert Motherwell",
      image: {
        id: "5b4b8703a7bb3b324e71bb0b",
        url: "https://d32dm0rphc51dk.cloudfront.net/u-Onw4lFPIgSERW9vkizjA/normalized.jpg",
      },
      price: "$6,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazplbGxzd29ydGgta2VsbHktZ3JlZW4tY3VydmUtMTQ=",
      id: "ellsworth-kelly-green-curve-14",
      artist_names: "Ellsworth Kelly",
      image: {
        id: "5b4b7c655a8a0c02f6df7999",
        url: "https://d32dm0rphc51dk.cloudfront.net/Cut7VPHWKBilZ5Ivj5vJIg/normalized.jpg",
      },
      price: "$12,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazp3b2xmLWthaG4taW4taXBzd2ljaC1tYXNz",
      id: "wolf-kahn-in-ipswich-mass",
      artist_names: "Wolf Kahn",
      image: {
        id: "5b4b7a60031bec5fa824e4d7",
        url: "https://d32dm0rphc51dk.cloudfront.net/qY9kBnLdh2EPkPZK8VuAaA/normalized.jpg",
      },
      price: "$10,500",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpkYXZpZC1ob2NrbmV5LW15LXBvb2wtYW5kLXRlcnJhY2UtNA==",
      id: "david-hockney-my-pool-and-terrace-4",
      artist_names: "David Hockney",
      image: {
        id: "5b4b7596c3f01104a3b8cd03",
        url: "https://d32dm0rphc51dk.cloudfront.net/05rL1DSRwbn9t43xoYkiLg/normalized.jpg",
      },
      price: "$10,500",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazp3b2xmLWthaG4tb3JjaGFyZC1hdC1kdXNr",
      id: "wolf-kahn-orchard-at-dusk",
      artist_names: "Wolf Kahn",
      image: {
        id: "5b4b7a3fc4c7147ec8ed12b8",
        url: "https://d32dm0rphc51dk.cloudfront.net/7YGi2ogpyQW4Vl9gorV0dQ/normalized.jpg",
      },
      price: "$9,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpoYW5zLWhvZm1hbm4tdW50aXRsZWQtMjIx",
      id: "hans-hofmann-untitled-221",
      artist_names: "Hans Hofmann",
      image: {
        id: "5b4b7872841d6662f71bb141",
        url: "https://d32dm0rphc51dk.cloudfront.net/WPV9hXX5oms437ujKpPdGg/normalized.jpg",
      },
      price: "$8,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpwb2xseS1hcGZlbGJhdW0tbG92ZS1tZS10ZW5kZXItaWk=",
      id: "polly-apfelbaum-love-me-tender-ii",
      artist_names: "Polly Apfelbaum",
      image: {
        id: "5b4cccb8efdaeb002387b941",
        url: "https://d32dm0rphc51dk.cloudfront.net/HPFxUT7W1lV6UwRu3PvlRQ/normalized.jpg",
      },
      price: "$8,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpicmlkZ2V0LXJpbGV5LWJsdWUtZG9taW5hbmNl",
      id: "bridget-riley-blue-dominance",
      artist_names: "Bridget Riley",
      image: {
        id: "5b4b64635a8a0c796e7ee64f",
        url: "https://d32dm0rphc51dk.cloudfront.net/SvWPlpYA25k-5Kje-Lp8rg/normalized.jpg",
      },
      price: "$35,000",
      availability: "for sale",
      contact_label: "Gallery",
    },
    {
      __id:
        "QXJ0d29yazpwYWJsby1waWNhc3NvLW1vZGVsLWNvbnRlbXBsYXRpbmctYS1ncm91cC1vZi1zY3VscHR1cmVzLW1vZGVsZS1jb250ZW1wbGFudC11bi1ncm91cGUtc2N1bHB0ZQ==",
      id: "pablo-picasso-model-contemplating-a-group-of-sculptures-modele-contemplant-un-groupe-sculpte",
      artist_names: "Pablo Picasso",
      image: {
        id: "5b4b9901640675075eedee14",
        url: "https://d32dm0rphc51dk.cloudfront.net/leU66mGhe0D2OvMQmGRt4Q/large_rectangle.jpg",
      },
      price: "",
      availability: "sold",
      contact_label: "Gallery",
    },
    {
      __id: "QXJ0d29yazpkYWxlLWNoaWh1bHktY2VsdGljLWVtZXJhbGQtcGVyc2lhbi1wYWly",
      id: "dale-chihuly-celtic-emerald-persian-pair",
      artist_names: "Dale Chihuly",
      image: {
        id: "5b4b9d53a7bb3b3b27bcec67",
        url: "https://d32dm0rphc51dk.cloudfront.net/6hN0QtT-BQE2y8LGhiI3EA/normalized.jpg",
      },
      price: "",
      availability: "sold",
      contact_label: "Gallery",
    },
  ],
  counts: {
    artworks: 21,
    eligible_artworks: 21,
  },
  partner: {
    name: "Anderson Fine Art Gallery",
  },
  name: "Flickinger Collection",
  description:
    "Masterworks on paper, prints, paintings, and sculptures by mainstay postwar and contemporary artists are the focus in this exclusive exhibition showcasing the discerning connoisseurship of Tom & Penny Flickinger, lifelong collectors, and trustees of the Albright Knox Art Gallery.",
  press_release:
    "Paintings and Sculpture from the Sea Island Estate of the Flickingers will be offered for sale through the Anderson Fine Art Gallery, online and by appointment. Acquired over a 40 year period, the collection consists of post-war to contemporary paintings and works on paper.  The artwork was purchased from major dealers in New York, Chicago and London and reflects the passion, education and knowledge of Tom and Penny Flickinger whose intellectual discovery and visual pleasure guided their choices. \r\n\r\nThe collection includes works by Richard Diebenkorn, David Hockney, Joan Mitchell and Wolf Kahn, to name a few. A portion of the proceeds will be donated to the St. Simons Land Trust.",
  exhibition_period: "Jul 1 – Oct 30",
  status_update: "Closing tomorrow",
  images: [
    {
      url: "https://d32dm0rphc51dk.cloudfront.net/Bu7ONuoTlR-PLZDUjs6ZTA/medium.jpg",
      aspect_ratio: 0.87,
    },
  ],
  filteredArtworks: {
    aggregations: [
      {
        slice: "PRICE_RANGE",
        counts: [
          {
            id: "*-*",
            name: "for Sale",
          },
          {
            id: "10000-50000",
            name: "between $10,000 & $50,000",
          },
          {
            id: "5000-10000",
            name: "between $5,000 & $10,000",
          },
          {
            id: "1000-5000",
            name: "between $1,000 & $5,000",
          },
          {
            id: "*-1000",
            name: "Under $1,000",
          },
        ],
      },
      {
        slice: "MEDIUM",
        counts: [
          {
            id: "painting",
            name: "Painting",
          },
          {
            id: "prints",
            name: "Prints",
          },
          {
            id: "work-on-paper",
            name: "Work on Paper",
          },
          {
            id: "drawing",
            name: "Drawing",
          },
          {
            id: "sculpture",
            name: "Sculpture",
          },
        ],
      },
    ],
    artworks_connection: {
      edges: [
        {
          node: {
            id: "david-hockney-my-pool-and-terrace-4",
            title: "My Pool & Terrace",
            date: "1983",
            sale_message: "$13,000",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/05rL1DSRwbn9t43xoYkiLg/large.jpg",
              aspect_ratio: 1.48,
            },
            artists: [
              {
                name: "David Hockney",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/david-hockney-my-pool-and-terrace-4",
          },
        },
        {
          node: {
            id: "bridget-riley-blue-dominance",
            title: "Blue Dominance",
            date: "1977",
            sale_message: "$15,000",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/SvWPlpYA25k-5Kje-Lp8rg/large.jpg",
              aspect_ratio: 0.5,
            },
            artists: [
              {
                name: "Bridget Riley",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/bridget-riley-blue-dominance",
          },
        },
        {
          node: {
            id: "hans-hofmann-untitled-221",
            title: "Untitled",
            date: "1942",
            sale_message: "$5,000",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/WPV9hXX5oms437ujKpPdGg/large.jpg",
              aspect_ratio: 0.82,
            },
            artists: [
              {
                name: "Hans Hofmann",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/hans-hofmann-untitled-221",
          },
        },
        {
          node: {
            id: "wolf-kahn-orchard-at-dusk",
            title: "Orchard at Dusk",
            date: "1989",
            sale_message: "$8,000",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/7YGi2ogpyQW4Vl9gorV0dQ/large.jpg",
              aspect_ratio: 1.25,
            },
            artists: [
              {
                name: "Wolf Kahn",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/wolf-kahn-orchard-at-dusk",
          },
        },
        {
          node: {
            id: "wolf-kahn-in-ipswich-mass",
            title: "In Ipswich, Mass",
            date: "1987",
            sale_message: "$9,500",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/qY9kBnLdh2EPkPZK8VuAaA/large.jpg",
              aspect_ratio: 1.35,
            },
            artists: [
              {
                name: "Wolf Kahn",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/wolf-kahn-in-ipswich-mass",
          },
        },
        {
          node: {
            id: "marc-hanson-scrappy",
            title: "Scrappy",
            date: "ca. 2018",
            sale_message: "$10,300",
            is_in_auction: false,
            is_biddable: false,
            is_acquireable: false,
            sale: null,
            sale_artwork: null,
            image: {
              url: "https://d32dm0rphc51dk.cloudfront.net/GSzNrBCUurMHdNEgT6LdwQ/large.jpg",
              aspect_ratio: 1.33,
            },
            artists: [
              {
                name: "Marc Hanson",
              },
            ],
            partner: {
              name: "Anderson Fine Art Gallery",
            },
            href: "/artwork/marc-hanson-scrappy",
          },
        },
      ],
    },
  },
  artists_grouped_by_name: [
    {
      letter: "A",
      items: [
        {
          id: "polly-apfelbaum",
          __id: "QXJ0aXN0OnBvbGx5LWFwZmVsYmF1bQ==",
          name: "Polly Apfelbaum",
          is_followed: false,
          nationality: "American",
          birthday: "1955",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/MByqc96STE-fJtW4mZpgSg/tall.jpg",
          },
        },
        {
          id: "karel-appel",
          __id: "QXJ0aXN0OmthcmVsLWFwcGVs",
          name: "Karel Appel",
          is_followed: false,
          nationality: "Dutch",
          birthday: "1921",
          deathday: "2006",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/puxzysW1yY3FCRLFaLFEQQ/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "B",
      items: [
        {
          id: "bruce-beasley",
          __id: "QXJ0aXN0OmJydWNlLWJlYXNsZXk=",
          name: "Bruce Beasley",
          is_followed: false,
          nationality: "American",
          birthday: "1939",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/XazCw3_7ZEQz-sdkLDWXjA/tall.jpg",
          },
        },
        {
          id: "nell-blaine",
          __id: "QXJ0aXN0Om5lbGwtYmxhaW5l",
          name: "Nell Blaine",
          is_followed: false,
          nationality: "American",
          birthday: "1922",
          deathday: "1996",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/jr4zSDi7sXllVBNVhQardw/tall.jpg",
          },
        },
        {
          id: "carolyn-brady",
          __id: "QXJ0aXN0OmNhcm9seW4tYnJhZHk=",
          name: "Carolyn Brady",
          is_followed: false,
          nationality: "American",
          birthday: "1937",
          deathday: "2005",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/DH5Jtvy_a5d4Bb5hxB-Fxw/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "C",
      items: [
        {
          id: "dale-chihuly",
          __id: "QXJ0aXN0OmRhbGUtY2hpaHVseQ==",
          name: "Dale Chihuly",
          is_followed: false,
          nationality: "American",
          birthday: "1941",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/qfT1qZw8EGJguSakyhFxig/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "D",
      items: [
        {
          id: "richard-diebenkorn",
          __id: "QXJ0aXN0OnJpY2hhcmQtZGllYmVua29ybg==",
          name: "Richard Diebenkorn",
          is_followed: false,
          nationality: "American",
          birthday: "1922",
          deathday: "1993",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/C5GfncAzGMsMf6A6RSfJEQ/tall.jpg",
          },
        },
        {
          id: "jim-dine",
          __id: "QXJ0aXN0OmppbS1kaW5l",
          name: "Jim Dine",
          is_followed: false,
          nationality: "American",
          birthday: "1935",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/Ll4gMf_OsLMehbgv05H6mw/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "F",
      items: [
        {
          id: "sam-francis",
          __id: "QXJ0aXN0OnNhbS1mcmFuY2lz",
          name: "Sam Francis",
          is_followed: false,
          nationality: "American",
          birthday: "1923",
          deathday: "1994",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/wZ9dabmVZjtcaB_q_bKKzw/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "H",
      items: [
        {
          id: "robert-harms",
          __id: "QXJ0aXN0OnJvYmVydC1oYXJtcw==",
          name: "Robert Harms",
          is_followed: false,
          nationality: "",
          birthday: "1962",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/i5_X3qJO8gFiyHrlHvAb-w/tall.jpg",
          },
        },
        {
          id: "david-hockney",
          __id: "QXJ0aXN0OmRhdmlkLWhvY2tuZXk=",
          name: "David Hockney",
          is_followed: false,
          nationality: "British",
          birthday: "1937",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/kCJVZo7bcqVrjnQ22QHhvg/tall.jpg",
          },
        },
        {
          id: "hans-hofmann",
          __id: "QXJ0aXN0OmhhbnMtaG9mbWFubg==",
          name: "Hans Hofmann",
          is_followed: false,
          nationality: "German-American",
          birthday: "1880",
          deathday: "1966",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/vRQdNHjLuWKadBXy9Fl4Pg/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "K",
      items: [
        {
          id: "wolf-kahn",
          __id: "QXJ0aXN0OndvbGYta2Fobg==",
          name: "Wolf Kahn",
          is_followed: false,
          nationality: "American",
          birthday: "1927",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/pzYQSjFUn7MyELRIVyADjQ/tall.jpg",
          },
        },
        {
          id: "ellsworth-kelly",
          __id: "QXJ0aXN0OmVsbHN3b3J0aC1rZWxseQ==",
          name: "Ellsworth Kelly",
          is_followed: false,
          nationality: "American",
          birthday: "1923",
          deathday: "2015",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/Bv9Hh3cYTCvc76NnZS4w4g/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "M",
      items: [
        {
          id: "robert-motherwell",
          __id: "QXJ0aXN0OnJvYmVydC1tb3RoZXJ3ZWxs",
          name: "Robert Motherwell",
          is_followed: false,
          nationality: "American",
          birthday: "1915",
          deathday: "1991",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/BbPC9SljRn-_6AfewiQpxw/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "P",
      items: [
        {
          id: "pablo-picasso",
          __id: "QXJ0aXN0OnBhYmxvLXBpY2Fzc28=",
          name: "Pablo Picasso",
          is_followed: false,
          nationality: "Spanish",
          birthday: "1881",
          deathday: "1973",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/i3rCA3IaKE-cLBnc-U5swQ/tall.jpg",
          },
        },
      ],
    },
    {
      letter: "R",
      items: [
        {
          id: "bridget-riley",
          __id: "QXJ0aXN0OmJyaWRnZXQtcmlsZXk=",
          name: "Bridget Riley",
          is_followed: false,
          nationality: "British",
          birthday: "1931",
          deathday: "",
          image: {
            url: "https://d32dm0rphc51dk.cloudfront.net/zLIFDjlzCw8ijOH9PPeasA/tall.jpg",
          },
        },
      ],
    },
  ],
  events: [
    {
      event_type: "Opening Reception",
      description: "",
      title: "",
      start_at: "2018-11-02T18:00:00+00:00",
      end_at: "2018-11-02T21:00:00+00:00",
    },
    {
      event_type: "Guided Tour",
      description: "Keith Macgregor will take visitors around the gallery and share his stories.",
      title: "",
      start_at: "2018-11-10T11:00:00+00:00",
      end_at: "2018-11-10T12:00:00+00:00",
    },
    {
      event_type: "Artist Talk",
      description: "Meet and greet Keith Macgregor who will be glad to share his stories and answer your questions.",
      title: "",
      start_at: "2018-11-10T14:00:00+00:00",
      end_at: "2018-11-10T18:00:00+00:00",
    },
  ],
}
