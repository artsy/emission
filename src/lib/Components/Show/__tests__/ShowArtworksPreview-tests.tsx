import { ShowFixture } from "lib/__fixtures__/ShowFixture"
import { MockRelayRenderer } from "lib/tests/MockRelayRenderer"
import { renderUntil } from "lib/tests/renderUntil"
import React from "react"
import { graphql } from "react-relay"
import { ShowArtworksPreviewContainer as ShowArtworksPreview } from "../ShowArtworksPreview"

jest.unmock("react-relay")

it("renders properly", async () => {
  const tree = await renderUntil(
    wrapper => {
      return wrapper.text().includes("All works")
    },
    <MockRelayRenderer
      Component={(props: any) => <ShowArtworksPreview title="All works" {...props} />}
      query={graphql`
        query ShowArtworksPreviewTestsQuery {
          show(id: "anderson-fine-art-gallery-flickinger-collection") {
            ...ShowArtworksPreview_show
          }
        }
      `}
      mockResolvers={{
        Show: () => ShowFixture,
      }}
    />
  )

  expect(tree.html()).toMatchSnapshot()
})
