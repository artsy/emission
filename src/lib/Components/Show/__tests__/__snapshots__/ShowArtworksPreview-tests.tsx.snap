// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `"<text font-family=\\"AGaramondPro-Regular\\" font-size=\\"26px\\" lineheight=\\"32px\\" class=\\"sc-EHOje jvzENE\\" mt=\\"2\\" mb=\\"3\\">All works</text>"`;
